src/components/Bookings/Bookings.jsx

```diff
@@ -63,31 +64,71 @@ const Bookings = () => {
       )
     );
   };
+  //error handling for the form inputs
+  const [error, setError] = useState(null);
+
+  // Function to check the email format
+  const isValidEmail = (email) => {
+    const emailParts = email.split("@");
+    if (emailParts.length !== 2) {
+      return false;
+    }
+
+    const [localPart, domainPart] = emailParts;
+    if (localPart.length === 0 || domainPart.length === 0) {
+      return false;
+    }
+
+    const domainParts = domainPart.split(".");
+    if (domainParts.length < 2) {
+      return false;
+    }
+
+    return true;
+  };
 
   const bookingSubmit = (e) => {
     e.preventDefault();
+
+    // Check for empty values in all fields
+    for (const field in newBooking) {
+      if (!newBooking[field]) {
+        setError("All fields must be filled in.");
+        return;
+      }
+    }
     // Check if the room ID is already booked
     const isRoomAlreadyBooked = bookings.some(
       (booking) => booking.roomId === newBooking.roomId
     );
 
     if (isRoomAlreadyBooked) {
-      alert("This room is already booked. Please choose another room.");
+      setError("This room is already booked. Please choose another room.");
       return; // Prevent further execution if the room is already booked
     }
-    // Check if any of the required fields are empty
-    const requiredFields = [
-      "firstName",
-      "surname",
-      "email",
-      "roomId",
-      "checkInDate",
-      "checkOutDate",
-    ];
-    if (requiredFields.some((field) => !newBooking[field])) {
-      alert("Please fill in all required fields.");
+
+    // Validation for first name, last name, email, and room ID
+    if (
```

Comment:
That because it's only works on the HTML on react I'm trying to check if any of the fields are empty, that's what I understood when I was testing the validation function. It would make an extra line if I added the required attribute on the input fields